src/screens/lists/Lists.tsx

```diff
@@ -55,76 +96,287 @@ const Lists = () => {
+              getListData();
+            },
+          },
+        ],
+      );
+    }
+  };
+
+  const onPressListTitle = (list: ListInfo) => {
+    if (!addingCardToListId) {
+      setListTitle(list?.name);
+      setEditListById(list?.id);
+    }
+  };
+
+  const onPressAddCard = (id: string) => {
+    if (!editListbyId) setAddingCardToListId(id);
+  };
+
+  const onPressMove = (cardId: string, listId: string) => {
+    if (!addingCardToListId && !editListbyId && !isAddList) {
+      setIsVisible(true);
+      const list = lists?.find(({id}) => id === listId);
+      setSelectedList(list as ListInfo);
+      setSelectedCard(cardId);
+    }
+  };
+
+  const onSelectList = async (value: ListInfo) => {
+    setSelectedList(value);
+  };
+
+  const onPressModalClear = () => {
+    setIsVisible(false);
+  };
+  const onPressModalCheck = async () => {
+    await updateData(updateCardList(token, selectedCard, selectedList?.id));
+    getCardData();
+    setIsVisible(false);
+  };
+
+  const onPressCardDelete = (id: string) => {
+    if (!addingCardToListId && !editListbyId && !isAddList) {
+      Alert.alert('Delete Card', 'Are you sure you want to delete this card', [
+        {text: 'Cancel', style: 'cancel'},
+        {
+          text: 'Delete',
+          style: 'destructive',
+          onPress: async () => {
+            await deleteData(deletCardUrl(token, id));
+            getCardData();
+            ShowToast('success', 'Card deleted successfully');
+          },
+        },
+      ]);
+    }
   };
 
   const renderCard = ({item}: {item: CardInfo}) => (
     <TouchableOpacity onPress={() => onPressCard(item.id)}>
       <View style={styles.cardContainer}>
         <Text style={styles.cardTitle}>{item.name}</Text>
+        <View style={styles.cardActions}>
+          <Pressable
+            onPress={() => onPressMove(item?.id, item?.idList as string)}>
+            <HeaderIcon
+              name={'move-up'}
+              color={
+                addingCardToListId || editListbyId || isAddList
+                  ? colors.gray200
+                  : colors.blue700
+              }
+              size={20}
+            />
+          </Pressable>
+          <Pressable onPress={() => onPressCardDelete(item?.id)}>
+            <HeaderIcon
+              name={'delete'}
+              color={
+                addingCardToListId || editListbyId || isAddList
+                  ? colors.gray200
+                  : colors.red500
+              }
+              size={20}
+            />
+          </Pressable>
+        </View>
       </View>
     </TouchableOpacity>
   );
 
   const renderList = (list: ListInfo) => (
     <View key={list.id} style={styles.listContainer}>
       <View style={styles.listChildContainer}>
-        <Text style={styles.listTitle}>{list.name}</Text>
-        <FlatList data={cardsArr(list.id)} renderItem={renderCard} />
+        {editListbyId === list?.id ? (
+          <View style={styles.titleContainer}>
```

Comment:
extract these to separate render components, it is not clear what is rendering because of the nested conditions